src/graph_ops.rs

```diff
@@ -96,34 +96,46 @@ where
 {
     fn for_each_node<T, F>(
         &self,
-        partition: &[Range<Node>],
+        maybe_partition: Option<&[Range<Node>]>,
         node_values: &mut [T],
         node_fn: F,
     ) -> Result<(), Error>
     where
         T: Send,
         F: Fn(&Self, Node, &mut T) + Send + Sync,
     {
-        if partition.iter().map(|r| r.end - r.start).sum::<Node>() != self.node_count() {
-            return Err(Error::InvalidPartitioning);
```

Comment:
Can we remove the `Error::InvalidPartitioning` then?

Reply:
Well, we still need it for `for_each_node_by_partition` right?